ADD Support for the AROS (www.aros.org) operating system.

Kalamatee opened the pull request.

DanielGibson commented:
Hi, cool to see Dhewm3 being ported to a new platform! :-)

I wonder why you renamed dhewm3 to ADoom3 and add a generic icon for that (instead of using neo/sys/linux/setup/image/doom3.png)? I think it'd make sense if it's called dhewm3 on all platforms, also makes it easier for users to find the project homepage when they know the right name etc..

DanielGibson commented on base/mmakefile.src:
```diff
@@ -0,0 +1,18 @@
+# $Id$
```
if possible, all aros-specific files should reside in `neo/sys/aros/`

DanielGibson commented on neo/ADoom3.info.src:
```diff
@@ -0,0 +1 @@
+TYPE = DRAWER
```
same here (neo/sys/aros/)

and the following ADoom3.png I can't comment on because it's no text file - is it even needed here, it looks like you have the normal icons for doom3 and RoE in neo/sys/aros/ already?

DanielGibson commented on neo/framework/Licensee.h:
```diff
@@ -34,15 +34,23 @@ If you have questions concerning this license or the applicable additional terms
 */
 
+#if defined(__AROS__)
+#define GAME_NAME						"ADoom3"		// appears on window titles and errors
```
why all the renaming?

DanielGibson commented on neo/mmakefile.src:
```diff
@@ -0,0 +1,50 @@
+# Copyright � 2012-2017, Nick "Kalamatee" Andrews.
```
again: could this be in `neo/sys/aros/` instead?

Of that's really not possible, at least mention somewhere at the very beginning in a comment that this is for aros,
so people don't mistake it for a generic makefile

Kalamatee commented:
On 1 April 2017 at 23:16, Daniel Gibson <notifications@github.com> wrote:

> Hi, cool to see Dhewm3 being ported to a new platform! :-)
>

Thanks - it took a lot of work (mostly on the side of fixing AROS) to get
it ported and running. We have tested it with mesa 7.5 through to mesa 17
with softpipe, nouveaue and intelgma gallium drivers.

> I wonder why you renamed dhewm3 to ADoom3 and add a generic icon for that
> (instead of using neo/sys/linux/setup/image/doom3.png)? I think it'd make
> sense if it's called dhewm3 on all platforms, also makes it easier for
> users to find the project homepage when they know the right name etc..
>
ADoom3 was used mainly because we have an old Doom port called ADoom
(Amiga/AROS Doom) so it made sense to continue using the familiar name -
however it still displays that it is using dhewm3. Its not a big issue, but
people recognise/find it easier than dhewm which they know nothing about.

The icon was done quite a while ago to fit in with the existing AROS
gorilla based ones (dimension wise) - Also I don't think I managed to find
a ROE one?


> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dhewm/dhewm3/pull/167#issuecomment-290950695>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKwmvCfPBhW0jmMPfYdm8JBM3inumiUqks5rrsy6gaJpZM4MwlTI>
> .
>

DanielGibson commented:
I think you did find a RoE icon: https://github.com/dhewm/dhewm3/pull/167/files#diff-92d696e4dd804e6dc70f0db1a0ca07a8 ;-)

Kalamatee commented:
On 1 April 2017 at 23:55, Daniel Gibson <notifications@github.com> wrote:

> *@DanielGibson* commented on this pull request.
>
> Do you think it'd make sense to add something aros specific (build
> dependencies or sth) to the readme?
>
All the dependencies ship with AROS so it builds in our build system "out
of the box" for all our target platforms (arm, i386, x86_64) - we just need
to issue "make aros-doom3"


> ------------------------------
>
> In base/mmakefile.src
> <https://github.com/dhewm/dhewm3/pull/167#discussion_r109295621>:
>
> > @@ -0,0 +1,18 @@
> +# $Id$
>
> if possible, all aros-specific files should reside in neo/sys/aros/
>
I understand that and thanks for the advice - I can see about moving them.


> ------------------------------
>
> In neo/ADoom3.info.src
> <https://github.com/dhewm/dhewm3/pull/167#discussion_r109295631>:
>
> > @@ -0,0 +1 @@
> +TYPE = DRAWER
>
> same here (neo/sys/aros/)
>
> and the following ADoom3.png I can't comment on because it's no text file
> - is it even needed here, it looks like you have the normal icons for doom3
> and RoE in neo/sys/aros/ already?
> ------------------------------
>
> In neo/framework/Licensee.h
> <https://github.com/dhewm/dhewm3/pull/167#discussion_r109295687>:
>
> > @@ -34,15 +34,23 @@ If you have questions concerning this license or the applicable additional terms
>  ===============================================================================
>  */
>
> +#if defined(__AROS__)
> +#define GAME_NAME						"ADoom3"		// appears on window titles and errors
>
> why all the renaming?
>
This is the only place we have changed the vars so that the binary/window
title display ADoom3 (dhewm3 1.4.1)


> ------------------------------
>
> In neo/mmakefile.src
> <https://github.com/dhewm/dhewm3/pull/167#discussion_r109295741>:
>
> > @@ -0,0 +1,50 @@
> +# Copyright � 2012-2017, Nick "Kalamatee" Andrews.
>
> again: could this be in neo/sys/aros/ instead?
>
> Of that's really not possible, at least mention somewhere at the very
> beginning in a comment that this is for aros,
> so people don't mistake it for a generic makefile
>

I will see about moving them.

Kalamatee commented:
On 1 April 2017 at 23:57, Daniel Gibson <notifications@github.com> wrote:

> I think you did find a RoE icon: https://github.com/dhewm/
> dhewm3/pull/167/files#diff-92d696e4dd804e6dc70f0db1a0ca07a8 ;-)
>
Sorry for not being clear - I meant I didn't think I found it in the source
tree for dhewm3 - it was a few years ago now though so my memory is a bit
hazy ;p

Kalamatee commented:
On 2 April 2017 at 00:01, Kalamatee <kalamatee@gmail.com> wrote:

>
>
> On 1 April 2017 at 23:57, Daniel Gibson <notifications@github.com> wrote:
>
>> I think you did find a RoE icon: https://github.com/dhewm/dhewm
>> 3/pull/167/files#diff-92d696e4dd804e6dc70f0db1a0ca07a8 ;-)
>>
> Sorry for not being clear - I meant I didn't think I found it in the
> source tree for dhewm3 - it was a few years ago now though so my memory is
> a bit hazy ;p
>
>
Ok.

I've had to leave our top level mmakefile.src in neo since it needs to be
there to correctly invoke cmake for our build system - but I have added a
comment to clarify that it is an aros specific file. I have moved the other
AROS specific files (folder icon and mmakefile.src for the default.cfg)
under neo/sys/aros/setup

I hope that is acceptable?

Thanks again for your input/reply.

Kalamatee commented:
On 2 April 2017 at 00:41, Kalamatee <kalamatee@gmail.com> wrote:

>
>
> On 2 April 2017 at 00:01, Kalamatee <kalamatee@gmail.com> wrote:
>
>>
>>
>> On 1 April 2017 at 23:57, Daniel Gibson <notifications@github.com> wrote:
>>
>>> I think you did find a RoE icon: https://github.com/dhewm/dhewm
>>> 3/pull/167/files#diff-92d696e4dd804e6dc70f0db1a0ca07a8 ;-)
>>>
>> Sorry for not being clear - I meant I didn't think I found it in the
>> source tree for dhewm3 - it was a few years ago now though so my memory is
>> a bit hazy ;p
>>
>>
> Ok.
>
> I've had to leave our top level mmakefile.src in neo since it needs to be
> there to correctly invoke cmake for our build system - but I have added a
> comment to clarify that it is an aros specific file. I have moved the other
> AROS specific files (folder icon and mmakefile.src for the default.cfg)
> under neo/sys/aros/setup
>
> I hope that is acceptable?
>
> Thanks again for your input/reply.
>
>
>
Hi Daniel,

Should I create a new pull request?

DanielGibson commented:
no need for a new pull request, I just haven't gotten around to look at it again

Kalamatee commented:
On 2 April 2017 at 17:57, Daniel Gibson <notifications@github.com> wrote:

> no need for a new pull request, I just haven't gotten around to look at it
> again
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dhewm/dhewm3/pull/167#issuecomment-290998934>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKwmvIqp6_3_yLgOlt3DdzGDtS_e_Z4Wks5rr9NxgaJpZM4MwlTI>
> .
>
N.P - and thanks again.

DanielGibson commented:
btw, I was told there's already a project named adoom3 for android.. are you sure you wanna use that name? ;)

Kalamatee commented:
On 3 April 2017 at 17:31, Daniel Gibson <notifications@github.com> wrote:

> btw, I was told there's already a project named adoom3 for android.. are
> you sure you wanna use that name? ;)
>

yes, ADoom has been used for the AmigaOS ports since '97 - shame they never
bothered to check first. (we also used it since 2012 for this port)

Kalamatee commented:
On 3 April 2017 at 17:41, Kalamatee <kalamatee@gmail.com> wrote:

>
>
> On 3 April 2017 at 17:31, Daniel Gibson <notifications@github.com> wrote:
>
>> btw, I was told there's already a project named adoom3 for android.. are
>> you sure you wanna use that name? ;)
>>
>
> yes, ADoom has been used for the AmigaOS ports since '97 - shame they
> never bothered to check first. (we also used it since 2012 for this port)
>

To clarify - the project is still dhewm3, just the AROS binary/port is
named ADoom3 to be consistent with our previous doom versions, and since
its already been reviewed as that since ~2013. Here's a relatively new(ish)
post of it on AROS https://www.youtube.com/watch?v=cXlO_lAxLbw

DanielGibson commented:
somehow the commit history is screwed up.. e.g. there are 3 occurences of the "use socklen_t" and "use the correct extension for x86" commits

Kalamatee commented:
On 10 April 2017 at 21:57, Daniel Gibson <notifications@github.com> wrote:

> somehow the commit history is screwed up.. e.g. there are 3 occurences of
> the "use socklen_t" and "use the correct extension for x86" commits
>
>
> Tell me about it - I unwisely decided to use tortoisegit. The duplicates
in the messages are from when it has told me I have had to do a special
commit to fix some conflict.

DanielGibson commented:
weird.. the resolving should be in a merge commit or maybe in the original commit that's changed (in case of rebasing), but that shouldn't create duplicate commits.. but I've never used tortoisegit so no idea what it does

DanielGibson commented:
So I take it you're not super familiar with git?

No worries, I'll try to clean up the commit history in the next days (with interactive rebase it shouldn't be too hard to remove duplicate commits and fix the merge conflicts in the then remaining commits)

Kalamatee commented:
On Mon, 10 Apr 2017 at 22:14, Daniel Gibson <notifications@github.com>
wrote:

> So I take it you're not super familiar with git?
>

No - I tend to use svn still mostly

> No worries, I'll try to clean up the commit history in the next days (with
> interactive rebase it shouldn't be too hard to remove duplicate commits and
> fix the merge conflicts in the then remaining commits)
>

Thank you very much.

—
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dhewm/dhewm3/pull/167#issuecomment-293081040>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKwmvMQIHPJWBXHehs73hp9FlU3WGIeDks5rupupgaJpZM4MwlTI>
> .
>

DanielGibson commented:
I cleaned up the history and merged it, thanks again :)
